src/transaction_guard.rs

```diff
@@ -0,0 +1,58 @@
+use std::sync::atomic::Ordering;
+
+use crate::Environment;
+
+#[must_use = "TransactionGuard has no effect without holding its object"]
+pub struct TransactionGuard<'a> {
+    env: &'a Environment,
+}
+
+/// A transaction guard is a static global counter that keeps track of all transactions currently open
+/// It can be used for operations that need to wait for all transactions to be closed, such as map-resize
+impl<'a> TransactionGuard<'a> {
+    pub fn new(env: &'a Environment) -> Self {
+        ScopedTransactionBlocker::wait_for_all_blockers(env);
+        env.tx_count().fetch_add(1, Ordering::Relaxed);
```

Comment:
I am not convinced this works in all cases. Suppose we wait for all blockers here, then context switches to something that starts a new "blocker" so we have \blocker_count > 0, then the context switches back here and a transaction is allowed to proceed while a blocker is active.

Or did I miss something? All the spin locks make my head spin...

Reply:
I added a spinlock to prevent that race. I wonder whether we could abstract all this in one thing. I'll have to think about it.